compiler/runtime/OMRAOTRelocationRuntime.cpp

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "runtime/OMRAOTRelocationRuntime.hpp"
+#include <string>
+
+TR::RelocationRuntime * OMR::OMRAOTRelocationRuntime::self()
+   {
+   return reinterpret_cast<TR::RelocationRuntime*>(this);
+   }
+
+void *
+OMR::OMRAOTRelocationRuntime::symbolAddress(char *symbolName)
```

Comment:
Hello @mstoodle , thank you for your question. At this stage, we've never used ResolvedMethod (we simply do not store as much information in OMR AOT atm).  Those just keep track of loaded method code and addresses.  We're also keying it my method name in several other places, like https://github.com/eclipse/omr/pull/4557/files/d26eb009b923aadb8e40eddaad0c3ed0fd5c8482#diff-1cd958730087bceee450395254da114dR101-R110 .  Since you have pointed that out, could you please suggest what improvements are desired? Shall we actually instantiate ResolvedMethods with missing data when loading? Is there an advice you could give to help us with the design of the improvement?

Reply:
Typically, I think a project extending OMR would define its own ResolvedMethod class (just like JitBuilder does). I would have preferred this approach be taken rather than falling back on something like a string which, while general, might require complex mangling to differentiate methods that are not the same thing (even though they may have the same name or some part of its signature may be the same).

Reply:
Though come to think of it, this particular one is a "symbol" which I presume may not be a method? It could be a data symbol?

Reply:
Thank you for the comment @mstoodle.

To give you some more context:
Relocation Runtime in our design ended up being aware of code addresses that are needed to be patched. 
We have been storing loaded addresses in there to query for them from relocation records, for instance when a method needed to call another method. 
So far, we have used it just for method addresses, but my next research item will need to persist and operate with some data.
I have been using the word "symbol" to mean "anything that is loaded", I've come to realize the name "persistedItem" is more generic and might serve this purpose better, i.e. does not promise to be a proper method name.

I do agree the provided code is far from perfect, especially when doing string comparison for methods (the problem with types and names is very real), so I am suggesting the following:
There should be more thought put into separating kinds of persisted items (i.e Resolved Methods, maybe some ResolvedDataItems), but we've never needed anything beyond a pointer to code memory location so far. 
Another thing, I still cannot wrap my thoughts (probably because of the lack of expertise) to predict what needs to be done to recreate a Resolved Method in some cases.
For example, in the cases when we were linking with a function loaded from ELF, special demangling procedure should be done.
It is also a bit unclear where the ResolvedMethod should be created, since loading is currently instantiated on the runtime side.
For the two reasons above, I think the resolved method should be discussed along with  interacting with JITBuilder in #5119, because a request to load a resolved method can be issued by JITBuilder, that has a ResolvedMethod in hand, or by a runtime environment consuming OMR someplace else.
Also, I am suggesting having a discussion about more different methods appearing in the AOTRelocationRuntime in future, having it noted down in #5469.
Currently, I have reworded the methods to be working with "items" as the minimally working prototype.

What do you think about the changes, and shall the discussion be continued in the other issues?